File: RobolectricDemo/app/src/main/java/org/lynxz/robolectricdemo/BaseActivity.java
package org.lynxz.robolectricdemo;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.Toast;

/**
 * Created by zxz on 2016/7/11.
 */
public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        initData();
        initView();
        initListener();
    }

    protected abstract int getLayoutId();

    protected abstract void initData();

    protected abstract void initView();

    protected abstract void initListener();


    public <T extends View> T findView(int id) {
        return (T) super.findViewById(id);
    }

    public void showToast(String msg) {
        if (!TextUtils.isEmpty(msg)) {
            Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
        }
    }

    public void showToast(int id) {
        Toast.makeText(this, id, Toast.LENGTH_SHORT).show();
    }
}

File: RecyclerViewDemo/app/src/main/java/lynxz/org/recyclerviewdemo/BaseActivity.kt
package lynxz.org.recyclerviewdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * Created by Lynxz on 2016/11/16.
 * description :
 */
abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        init()
    }

    /**
     * 初始化数据
     * */
    abstract fun init()

    /**
     * 设置布局id
     * */
    abstract fun getLayoutId(): Int
}